Spotify/module.php

```diff
@@ -424,6 +535,45 @@ public function SetShuffle(bool $Shuffle)
             $this->SetValue('Shuffle', $Shuffle);
         }
 
+        public function UpdatePosition()
+        {
+            if ($this->ReadAttributeString('Token') != '') {
+                $currentPlay = $this->requestCurrentPlay();
+                if (empty($currentPlay)) {
+                    $this->SetValue('Position', 0);
+                } else {
+                    $current_volume = $currentPlay['device']['volume_percent'];
+                    $this->SetValue('Volume', $current_volume);
+                    $this->SendDebug('Volume', $current_volume, 0);
+
+                    $progress_ms = $currentPlay['progress_ms'];
+                    $this->SendDebug('Progress ms', $progress_ms, 0);
+                    $this->SendDebug('Current Position ms', $progress_ms, 0);
+                    $progress = $this->ConvertTimeInterval($progress_ms);
+                    $this->SendDebug('Progress', $progress, 0);
+                    // $album = $currentPlay['item']['album'];
+
+                    $duration_ms = $currentPlay['item']['duration_ms'];
+                    $this->SendDebug('Duration ms', $duration_ms, 0);
+                    $duration = $this->ConvertTimeInterval($duration_ms);
+                    $this->SendDebug('Duration', $duration, 0);
+                    $this->SetValue('Duration', $duration);
+
+                    $remaining_time = $this->GetRemainingTime($progress_ms, $duration_ms);
```

Comment:
Da diese Funktion nur hier genutzt wird, kannst du sie auch direkt hier anwenden und musst das nicht in eine Funktion auslagern